### PR TITLE
Add RVO Avoidance

### PR DESCRIPTION
**Pull request by arturocuma**
Add RVO Avoidance to be able to use AI Navigation for robot of different sizes
Bringing changes from Duc's branch AddRVOAvoidanceFloatMove
https://github.com/rapyuta-robotics/RapyutaSimulationPlugins/tree/AddRVOAvoidanceFloatMove

### CODE REVIEW

File: Source/RapyutaSimulationPlugins/Private/Drives/RRFloatingMovementComponent.cpp [hunk 1]
```diff
@@ -269,3 +298,149 @@ void URRFloatingMovementComponent::StopMovementImmediately()
...
+                    //DrawDebugLine(GetWorld(), GetActorLocation(), GetActorLocation() + Velocity, FColor::Green, false, 0.05f, SDPG_MAX, 10.0f);
+                }
+#endif
+            }
+        }
+        RVOAvoidanceManager->UpdateRVO(this);
+
+        bRVOAvoidanceRecentlyUpdated = true;
+    }
+#if RAPYUTA_SIM_DEBUG
```
Comment (yuokamoto) [inline Source/RapyutaSimulationPlugins/Private/Drives/RRFloatingMovementComponent.cpp]: Do you need both RAPYUTA_SIM_DEBUG and bShowDebug?

Comment (Tadinu) [reply Source/RapyutaSimulationPlugins/Private/Drives/RRFloatingMovementComponent.cpp:426]: bShowDebug should actually be named bRVODebugEnabled

File: Source/RapyutaSimulationPlugins/Public/Drives/RRFloatingMovementComponent.h [hunk 2]
```diff
@@ -66,7 +70,27 @@ class RAPYUTASIMULATIONPLUGINS_API URRFloatingMovementComponent : public UFloati
 
     virtual void StopMovementImmediately() override;
 
+    UPROPERTY(VisibleAnywhere, BlueprintReadOnly)
```
Comment (Tadinu) [inline Source/RapyutaSimulationPlugins/Public/Drives/RRFloatingMovementComponent.h:75]:
```suggestion
    // RVO Avoidance
    // Ref: UCharacterMovementComponent
    UPROPERTY(VisibleAnywhere, BlueprintReadOnly)
```

**Review by Tadinu — APPROVED** [review]

File: Source/RapyutaSimulationPlugins/Private/Drives/RRFloatingMovementComponent.cpp [hunk 3]
```diff
@@ -269,3 +298,149 @@ void URRFloatingMovementComponent::StopMovementImmediately()
...
+    {
+        return;
+    }
+
+    if (GetOwner()->GetLocalRole() != ROLE_Authority)
+    {
+        return;
+    }
+
+#if RAPYUTA_SIM_DEBUG
```
Comment (Tadinu) [inline Source/RapyutaSimulationPlugins/Private/Drives/RRFloatingMovementComponent.cpp]: const bool bRVODebugEnabled = RAPYUTA_SIM_DEBUG && RVOAvoidanceManager->IsDebugEnabled(RVOAvoidanceUID);
then use only bRVODebugEnabled for checks below

Comment (arturocuma) [reply Source/RapyutaSimulationPlugins/Private/Drives/RRFloatingMovementComponent.cpp]: OK

File: Source/RapyutaSimulationPlugins/Private/Drives/RRFloatingMovementComponent.cpp (hunk 1 above)
Comment (arturocuma) [reply Source/RapyutaSimulationPlugins/Private/Drives/RRFloatingMovementComponent.cpp]: I will use Duc's suggestion:
const bool bRVODebugEnabled = RAPYUTA_SIM_DEBUG && RVOAvoidanceManager->IsDebugEnabled(RVOAvoidanceUID);
then use only bRVODebugEnabled for checks below

File: Source/RapyutaSimulationPlugins/Public/Drives/RRFloatingMovementComponent.h (hunk 2 above)
Comment (arturocuma) [reply Source/RapyutaSimulationPlugins/Public/Drives/RRFloatingMovementComponent.h:75]: OK

**Pull request merged**